core/src/error.rs

```diff
@@ -72,48 +71,40 @@ mod tests {
     /// test that we can convert an error to a string
     fn to_string() {
         let err = HolochainError::new("foo");
-        assert_eq!("ErrorGeneric(\"foo\")", err.to_string(),);
+        assert_eq!(r#"ErrorGeneric("foo")"#, err.to_string());
     }
 
     #[test]
     /// test that we can convert an error to valid JSON
     fn test_to_json() {
         let err = HolochainError::new("foo");
-        assert_eq!("{\"error\":\"foo\"}", err.to_json());
+        assert_eq!(r#"{"error":"foo"}"#, err.to_json());
     }
 
     #[test]
     /// smoke test new errors
     fn can_instantiate() {
         let err = HolochainError::new("borked");
-        if let HolochainError::ErrorGeneric(err_msg) = err {
-            assert_eq!(err_msg, "borked")
-        } else {
-            assert!(false)
-        }
+
+        assert_eq!(HolochainError::ErrorGeneric("borked".to_string()), err);
     }
 
     #[test]
     /// test errors as a result and destructuring
     fn can_raise_holochain_error() {
-        let result = raises_holochain_error(true);
-        match result {
-            Ok(_) => assert!(false),
-            Err(err) => match err {
-                HolochainError::ErrorGeneric(msg) => assert_eq!(msg, "borked"),
-                _ => assert!(false),
-            },
+        let err = raises_holochain_error(true).expect_err("should return an error when yes=true");
+
+        match err {
+            HolochainError::ErrorGeneric(msg) => assert_eq!(msg, "borked"),
+            _ => panic!("raises_holochain_error should return an ErrorGeneric"),
         };
     }
 
     #[test]
     /// test errors as a returned result
     fn can_return_result() {
         let result = raises_holochain_error(false);
-        let result = match result {
-            Ok(x) => x,
-            Err(_) => panic!(),
-        };
-        assert_eq!(result, ())
+
+        assert_eq!(result, Ok(()))
```

Comment:
You could just assert for `result.is_ok()`

Reply:
I don't actually think that's clearer. 
Consider [this case](http://play.rust-lang.org/?gist=1520c29b94800ec83018a60a59ef6ec0&version=stable&mode=debug&edition=2015) which outputs:
```
thread 'main' panicked at 'assertion failed: `(left == right)`
 left: `Err(4)`,
 right: `Ok(())`', src/main.rs:3:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```
versus
[this other case](http://play.rust-lang.org/?gist=f16b58a2c17f755c650be6237fe9db55&version=stable&mode=debug&edition=2015) which outputs:
```
thread 'main' panicked at 'assertion failed: result.is_ok()', src/main.rs:3:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```
I think that the first error message is more descriptive.

If you disagree I can change this though.

Reply:
We might be more flexible with the latter. The actual return value has no real significance here. But we want to know, when the method call failed.